src/guacd/daemon.c

```diff
@@ -275,6 +282,12 @@ int main(int argc, char* argv[]) {
     /* General */
     int retval;
 
+    /* Set default stack size of 8MB */
+    pthread_attr_t default_pthread_attr;
+    pthread_attr_init(&default_pthread_attr);
+    pthread_attr_setstacksize(&default_pthread_attr, GUACD_THREAD_STACK_SIZE);
+    pthread_setattr_default_np(&default_pthread_attr);
```

Comment:
I don't think we should rely on `_GNU_SOURCE` causing `pthread_setattr_default_np()` to be defined. There's no requirement that a libc implementation behave differently with `_GNU_SOURCE` defined, nor that it provide `pthread_setattr_default_np()` to those defining it.

Instead, we should:

1. Use configure tests to check whether `pthread_setattr_default_np()` is available, setting `_GNU_SOURCE` if necessary.
2. Make these calls only if `pthread_setattr_default_np()` is available.

If it's possible to check whether setting `LDFLAGS` options is sufficient to establish the expected stack size, we might want to do that, as well. Setting `LDFLAGS` is the mechanism we use to achieve this within the current Alpine build:

https://github.com/apache/guacamole-server/blob/47b9360d4678bb8e60f0850090ccade666255069/src/guacd-docker/bin/build-all.sh#L34-L36

Reply:
Okay, this should be addressed via rebase.